### DISCUSSION TITLE
Modbus bridge usecase - can I alter 1 register?

### DISCUSSION THREAD

**Discussion by chrgraf** [eModbus/eModbus#302] [Q&A] [answered]
Dear all,

thanks for this really cool lib.

Have several use cases, but in general I have two inverters and both poll my smart meter and I need to fake one value.
So idea was using the bridge-example in roughly such topology:

real-smartermeter(slave)——[esp32-as-wifi/rtu-bridge]————(modbus-rtu-master)inverter1
    Register 243=100.                                 Modified register 243=200

https://github.com/eModbus/eModbus/blob/master/examples/WiFi-RTU_Bridge/main.cpp

Wondering if someone can explain me how I could modify a single register only, while all other registers are passed-through nativly please.

later I need to see how I can manage my two inverters, as one is rtu and the other is tcp. But that’s another topic..

first I need a solution to natively relay any response from slave to inverter while being able to alter one register on the fly.

Many thanks

christian

**Comment by Miq1** [comment]
One way would be to register the respective function code and server ID to the bridge itself. In the function handling these requests you could check if the very register is affected. Then start a ``syncRequest()`` to the real server and manipulate its response accordingly.

This may be failing if the server is a slow one, as the function will run in the bridge worker task that will time out eventually.

**Comment by Miq1** [comment, marked as answer]
One more idea that might do better:
- set up a client that will poll the power meter and keep the relevant data as a shadow copy.
- add both a TCP and RTU server that will use the same function to respond to data requests. This function will serve from the shadow data and is able to alter whatever you need in the responses.

**Reply by chrgraf** [reply]
Dear Miq1,

many thanks. I will follow-up on this.

many thanks

christian
